tests/test_parse_load_args.py

```diff
@@ -0,0 +1,160 @@
+    assert parse(["genkey"]).verb == "genkey"
+    assert parse(["bump"]).verb == "bump"
+    assert parse(["serve"]).verb == "serve"
+    assert parse(["build"]).verb == "build"
+    assert parse(["shell"]).verb == "shell"
+    assert parse(["boot"]).verb == "boot"
+    assert parse(["qemu"]).verb == "qemu"
+    with pytest.raises(SystemExit):
+        parse(["invalid"])
+
+def test_os_distribution():
+    for dist in Distribution:
+        assert parse(["-d", dist.name]).distribution == dist
+
+    with pytest.raises((argparse.ArgumentError, SystemExit)):
+        parse(["-d", "invalidDistro"])
+    with pytest.raises((argparse.ArgumentError, SystemExit)):
+        parse(["-d"])
+
+    for dist in Distribution:
+        with cd_temp_dir():
+            config = Path("mkosi.default")
+            config.write_text(f"[Distribution]\nDistribution={dist}")
+            assert parse([]).distribution == dist
+
+def test_machine_id():
+    id = uuid.uuid4().hex
+    load_args = parse(["--machine-id", id])
+
+    assert load_args.machine_id == id
+    assert load_args.machine_id_is_fixed
+
+    with pytest.raises(MkosiException):
+        parse(["--machine-id", "notValidKey"])
+    with pytest.raises((argparse.ArgumentError, SystemExit)):
+        parse(["--machine-id"])
+
+    with cd_temp_dir():
+        config = Path("mkosi.default")
+        config.write_text(f"[Output]\nMachineID={id}")
+        load_args = parse([])
+        assert load_args.machine_id == id
+        assert load_args.machine_id_is_fixed
+
+    with cd_temp_dir():
+        config = Path("mkosi.default")
+        config.write_text("[Output]\nMachineID=")
+        with pytest.raises(MkosiException):
+            parse([])
+
+def test_hostname():
+    assert parse(["--hostname", "name"]).hostname == "name"
+    with pytest.raises(SystemExit):
+        parse(["--hostname", "name", "additional_name"])
+    with pytest.raises(SystemExit):
+        parse(["--hostname"])
+
+    with cd_temp_dir():
+        config = Path("mkosi.default")
+        config.write_text("[Output]\nHostname=name")
+        assert parse([]).hostname == "name"
+
+    with cd_temp_dir():
+        config = Path("mkosi.default")
+        config.write_text("[Output]\nHostname=")
+        config = Path("hostname.txt")
+        assert parse([]).hostname == ""
+
+def test_centos_brtfs():
+    with cd_temp_dir():
+        config = Path("mkosi.default")
+        for dist in (Distribution.centos, Distribution.centos_epel):
+            for release in range (2, 9):
+                config.write_text(
+                    textwrap.dedent(
+                        f"""
+                        [Distribution]
+                        Distribution={dist}
+                        Release={release}
+                        [Output]
+                        Format=gpt_btrfs
+                        """
+                    )
+                )
+                with pytest.raises(MkosiException, match=".CentOS.*btrfs"):
```

Comment:
The match here is wrong causing the CI to fail. I'm assuming you meant ".*CentOS.*btrfs"

Reply:
It was not actually the *, but rather the error message being completely different from what I thought it was.
Not sure how I let this one slip.

> AssertionError: Regex pattern '.CentOS.*btrfs' does not match 'Sorry, CentOS 2 does not support unified kernel images. You must use --without-unified-kernel-images.'

Will fix it to:
![image](https://user-images.githubusercontent.com/71611489/151409148-532e1215-a11b-4f42-998e-d5199e78c905.png)

Reply:
Thinking about it, its best to add the * in case the order of the words are ever changed.